gammapy/obs/observation.py

```diff
@@ -319,3 +339,569 @@ def select_observations(self, selection=None):
                 raise ValueError('Invalid selection type: {}'.format(selection_type))
 
         return obs_table
+
+
+def recover_units(array, as_units):
+    """Utility functin to recover units.
+
+    After some numpy operations, `~astropy.units.Quantity`-like objects
+    loose their units. This function shoul recover them.
+
+    Parameters
+    ----------
+    array : `~numpy.ndarray` or `~astropy.units.Quantity`-like
+        Array without units.
+    as_units : int, float or `~astropy.units.Quantity`-like
+        Structure to imitate the units.
+
+    Returns
+    -------
+    array : int, float or `~astropy.units.Quantity`-like
+        Array with units.
+    """
+    try:
+        return Quantity(np.array(array), as_units.unit)
+    except:
+        # return unmodified
+        return array
+
+
+class ObservationGroups(object):
+
+    """Observation groups.
+
+    Class to define observation groups useful for organizing observation
+    lists into groups of observations with similar properties. The
+    properties and their binning are specified via
+    `~gammapy.obs.ObservationGroupAxis` objects.
+
+    The class takes as input a list of `~gammapy.obs.ObservationGroupAxis`
+    objects and defines 1 group for each possible combination of the
+    bins defined in all axes. The groups are identified by a unique
+    ``GROUP_ID`` int value.
+
+    The definitions of the groups are internally  stored as a
+    `~astropy.table.Table` object, the
+    `~gammapy.obs.ObservationGroups.obs_groups_table` member.
+
+    The axis parameters should be either dimensionless or castable
+    into `~astropy.units.Quantity` objects.
+
+    For details on the grouping of observations in a list, please
+    refer to the `~gammapy.obs.ObservationGroups.group_observation_table`
+    method.
+
+    Parameters
+    ----------
+    obs_group_axes : `~gammapy.obs.ObservationGroupAxis`
+        List of observation group axes.
+
+    Examples
+    --------
+    Define an observation grouping:
```

Comment:
I would find it really helpful to see the output of the print statements below.
My suggestion would be to:
- Move this example to the high-level docs page
- Link to it from this docstring
- Choose an example with fewer groups (~ 10) so that it's possible to print the relevant results.